### PR TITLE
ChecksFinder: Refactor/Cleaning

### PR DESCRIPTION
**Pull request by Exempt-Medic**
## What is this fixing or adding?

* Updates the ChecksFinder docs
* Removes ChecksFinder from the `non_apworlds` set
* Deletes `options.py` in favor of just `options_dataclass = PerGameCommonOptions` (this updates to the new API)
* Simplifies creation of items and locations
* Simplifies rules setting
* Simplifies `fill_slot_data`
* Removed `topology_present` because the game is simple
* Rearranges `__init__.py` to be in call order
* Fixes import errors and `per_slot_randoms`, uses `"` instead of `'` and uses new helper functions

## How was this tested?

Generations, reading and comparing spoiler logs on the same seeds, playing games in multiworlds to completion, and viewing the website tracker. Also unit tests

### CODE REVIEW

**Review by qwint — APPROVED** [review]
changes are all reasonable, fixed all the things I had brought up in discord

mostly just did a code review
Did not test the non_apworlds change, lightly tested on 3.8/3.11, lightly generated, and tested the full refactors individually

Did get a strange error on testing py3.8 but I also was getting it on current main so unlikely it's an issue with this PR, the error for posterity: `FAILED test/programs/test_multi_server.py::TestResolvePlayerName::test_resolve - RuntimeError: There is no current event loop in thread 'MainThread'.`

**Review by SunCatMC — APPROVED** [review]
I approve these changes

File: worlds/checksfinder/__init__.py [hunk 1]
```diff
@@ -25,67 +25,51 @@ class ChecksFinderWorld(World):
...
     def fill_slot_data(self):
-        slot_data = self._get_checksfinder_data()
-        for option_name in checksfinder_options:
-            option = getattr(self.multiworld, option_name)[self.player]
-            if slot_data.get(option_name, None) is None and type(option.value) in {str, int}:
-                slot_data[option_name] = int(option.value)
-        return slot_data
+        return {
+            "world_seed": self.random.getrandbits(32),
+            "eed_name": self.multiworld.seed_name,
```
Comment (agilbert1412) [inline worlds/checksfinder/__init__.py:66]: `eed_name`?

**Review by agilbert1412 — CHANGES_REQUESTED** [review]

File: worlds/checksfinder/__init__.py (hunk 1 above)
Comment (Exempt-Medic) [reply worlds/checksfinder/__init__.py]: Fixed

**Review by agilbert1412 — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]
Waiting on an answer from alwaysintreble about whether `options_dataclass = PerGameCommonOptions` has bad implications somehow. It seems fine to me but he would know better than I

Otherwise this is good

**Pull request merged**